operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -0,0 +1,635 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointslice
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+
+	"github.com/sirupsen/logrus"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/util/workqueue"
+)
+
+var (
+	// sequentialLetters contains lower case alphabets without vowels and few numbers.
+	// skipped vowels and numbers [0, 1] to avoid generating controversial names.
+	sequentialLetters = []rune("bcdfghjklmnpqrstvwxyz2456789")
+)
+
+// cesTracker holds the desired state of CiliumEndpointSlice and list of ceps to be removed
+// in next sync with k8s-apiserver.
+type cesTracker struct {
+	// Mutex to protect cep insert/removal in ces and removedCEPs
+	backendMutex lock.RWMutex
+	// The desired state of ces object
+	ces *cilium_v2.CiliumEndpointSlice
+	// set of CEPs to be removed in the CES object in next sync.
+	removedCEPs map[string]struct{}
+	// number of CEPs inserted in a CES
+	cepInserted int64
+	// number of CEPs removed from a CES
+	cepRemoved int64
+	// CES insert time at workqueue
+	cesInsertedAt time.Time
+}
+
+// cesManager
+type cesManager interface {
```

Comment:
I suggest renaming this interface as the name is not clear on what its purpose is. Below is a suggestion but if you find a better name, then by all means use that.

```suggestion
// operations is an interface to all operations that a CES manager can perform. 
type operations interface {
```

Reply:
I'm okay with changing to `operations`.